citesphere/src/main/java/edu/asu/diging/citesphere/core/service/giles/impl/GilesUploadCheckerImpl.java

```diff
@@ -140,8 +140,22 @@ public void checkUploads() {
                     // Giles is still procoessing
                     logger.debug("Upload " + upload.getProgressId()
                             + " still being processed.");
-                    checkedUploads.add(upload);
-                    continue;
+                    ObjectMapper mapper = new ObjectMapper();
+                    String jsonBody = response.getBody();
+                    GilesCheckUploadResponse processed = new GilesCheckUploadResponse();
+                    try {
+                        processed = mapper.readValue(jsonBody, GilesCheckUploadResponse.class);
```

Comment:
I think bey default this would throw an exception if one of the fields wouldn't be supplied, would it? Making it pretty prone to errors down the road. I think you need to annotate the GilesCheckUploadResponse with something that allows missing properties.